django/db/models/base.py

```diff
@@ -1087,12 +1087,22 @@ def _save_table(
             ]
             forced_update = update_fields or force_update
             pk_val = self._get_pk_val(meta)
-            updated = self._do_update(
-                base_qs, using, pk_val, values, update_fields, forced_update
+            returning_fields = [
+                field for field in meta.db_returning_fields if field not in pk_fields
+            ]
```

Comment:
This could be optimized to only return the generated fields that are functionally dependent on `update_fields` when they are specified. Currently this will return all generated fields.